modules/sparql-client/src/main/scala/ch/epfl/bluebrain/nexus/commons/sparql/client/SparqlClient.scala

```diff
@@ -86,7 +87,23 @@ abstract class SparqlClient[F[_]]()(implicit F: MonadError[F, Throwable]) {
     }
   }
 
-  def query(query: String): F[ResultSet]
+  /**
+    * @param q the query to execute
+    * @return the unmarshalled result set of the provided query executed against the sparql endpoint
+    */
+  def queryRs(q: String)(implicit rs: HttpClient[F, ResultSet]): F[ResultSet] =
+    query(q)(rs)
+
+  /**
+    * Executes the argument ''query'' against the underlying sparql endpoint.
+    *
+    * @param q the query to execute
+    * @return the raw result of the provided query executed against the sparql endpoint
+    */
+  def queryRaw(q: String)(implicit rs: HttpClient[F, Json]): F[Json] =
+    query(q)(rs)
+
+  def query[A](query: String)(implicit rs: HttpClient[F, A]): F[A]
```

Comment:
docs?

Reply:
This one for some reason was documented on the implementation. Should it be repeated here as well? Usually I prefer documenting in the trait then in the implementation though...

Reply:
The docs on the contract are more important than the ones in the implementation.